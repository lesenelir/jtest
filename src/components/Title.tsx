import React from "react"

interface IProps {
  type: string
  title: string
}

function Title(props: IProps) {
  const {type, title} = props

  return (
    <>
      {type === 'large' && <h1 className={'text-3xl text-blue-200'}>{title}</h1>}
      {type === 'small' && <h2 className={'text-lg text-blue-600'}>{title}</h2>}
    </>
  )
}

export default Title
