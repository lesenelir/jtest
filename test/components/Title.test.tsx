import React from "react"
import {render} from "@testing-library/react"

import Title from "@/components/Title"

describe('Title', () => {
  it('render large title successfully', () => {
    const {baseElement} = render(<Title type={'large'} title={'Large Title'}/>)
    expect(baseElement).toMatchSnapshot()
  })

  it('render small title successfully', () => {
    const {baseElement} = render(<Title type={'small'} title={'Small Title'}/>)
    expect(baseElement).toMatchSnapshot()
  })
})
