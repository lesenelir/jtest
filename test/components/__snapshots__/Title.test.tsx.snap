// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title render large title successfully 1`] = `
<body>
  <div>
    <h1
      class="text-3xl text-blue-200"
    >
      Large Title
    </h1>
  </div>
</body>
`;

exports[`Title render small title successfully 1`] = `
<body>
  <div>
    <h2
      class="text-lg text-blue-600"
    >
      Small Title
    </h2>
  </div>
</body>
`;
